libs/deps/mas/merch-card.js

```diff
@@ -1552,6 +1552,10 @@ merch-card [slot='heading-xs'] {
     text-decoration: none;
 }
 
+merch-card .badge-heading {
+    max-width: 70%;
```

Comment:
If you think this approach is safer/better, I recommend to remove `adjustTitleWidth` method in `libs/features/mas/web-components/src/variants/variant-layout.js` and solely use this approach.